src/contacts/contact-two-frames.cpp

```diff
@@ -0,0 +1,227 @@
+  updateForceInequalityConstraints();
+  updateForceRegularizationTask();
+
+  // This contact has forceGenMat as 3x3 identity matrix, so it can be used only
+  // for emulating a bll joint between two frames The forces calculated will
+  // have only linear part (rotation will be unconstrained) So we need to set
+  // the appropriate mask for motion task (which can take into account rotation
+  // but we don't need it)
+  math::Vector motion_mask(6);
+  motion_mask << 1., 1., 1., 0., 0., 0.;
+  m_motionTask.setMask(motion_mask);
+}
+
+void ContactTwoFrames::useLocalFrame(bool local_frame) {
+  m_dummyMotionTask.useLocalFrame(local_frame);
+}
+
+void ContactTwoFrames::updateForceInequalityConstraints() {
+  Matrix B = Matrix::Identity(3, 3);  // Force "gluing" two frames together can
+                                      // be arbitrary in sign/direction
+  Vector lb = m_fMin * Vector::Ones(3);
+  Vector ub = m_fMax * Vector::Ones(3);
+
+  m_forceInequality.setMatrix(B);
+  m_forceInequality.setLowerBound(lb);
+  m_forceInequality.setUpperBound(ub);
+}
+
+double ContactTwoFrames::getNormalForce(ConstRefVector f) const { return 0.0; }
+
+const Matrix3x& ContactTwoFrames::getContactPoints() const {
+  return m_contactPoints;
+}
+
+void ContactTwoFrames::setRegularizationTaskWeightVector(ConstRefVector& w) {
+  m_weightForceRegTask = w;
+  updateForceRegularizationTask();
+}
+
+void ContactTwoFrames::updateForceRegularizationTask() {
+  typedef Eigen::Matrix<double, 3, 3> Matrix3;
+  Matrix3 A = Matrix3::Zero();
+  A.diagonal() = m_weightForceRegTask;
+  m_forceRegTask.setMatrix(A);
+  m_forceRegTask.setVector(A * m_fRef);
+}
+
+void ContactTwoFrames::updateForceGeneratorMatrix() {
+  m_forceGenMat.setIdentity();
+}
+
+unsigned int ContactTwoFrames::n_motion() const { return m_motionTask.dim(); }
+unsigned int ContactTwoFrames::n_force() const { return 3; }
+
+const Vector& ContactTwoFrames::Kp() {
+  m_Kp3 = m_motionTask.Kp().head<3>();
+  return m_Kp3;
+}
+
+const Vector& ContactTwoFrames::Kd() {
+  m_Kd3 = m_motionTask.Kd().head<3>();
+  return m_Kd3;
+}
+
+void ContactTwoFrames::Kp(ConstRefVector Kp) {
+  assert(Kp.size() == 3);
+  Vector6 Kp6;
+  Kp6.head<3>() = Kp;
+  m_motionTask.Kp(Kp6);
+}
+
+void ContactTwoFrames::Kd(ConstRefVector Kd) {
+  assert(Kd.size() == 3);
+  Vector6 Kd6;
+  Kd6.head<3>() = Kd;
+  m_motionTask.Kd(Kd6);
+}
+
+bool ContactTwoFrames::setContactNormal(ConstRefVector contactNormal) {
+  /*assert(contactNormal.size()==3);
+  if(contactNormal.size()!=3)
+    return false;
+  m_contactNormal = contactNormal;
+  updateForceInequalityConstraints();*/
+  return true;
+}
+
+bool ContactTwoFrames::setFrictionCoefficient(
+    const double frictionCoefficient) {
+  /*
+  assert(frictionCoefficient>0.0);
+  if(frictionCoefficient<=0.0)
+    return false;
+  m_mu = frictionCoefficient;
+  updateForceInequalityConstraints();
+  */
+  return true;
+}
+
+bool ContactTwoFrames::setMinNormalForce(const double minNormalForce) {
```

Comment:
The methods for setting min and max forces should be implemented as they are used